lua/rustaceanvim/dap.lua

```diff
@@ -102,21 +103,83 @@ local function generate_source_map()
       local new_map = {
         [compat.joinpath('/rustc/', commit_hash)] = compat.joinpath(rustc_sysroot, '/lib/rustlib/src/rust'),
       }
-      vim.tbl_extend('force', source_map, { new_map })
+      source_map = vim.tbl_extend('force', source_map, { new_map })
     end)
   end)
 end
 
----@param args RADebuggableArgs
-function M.start(args)
-  vim.notify('Compiling a debug build for debugging. This might take some time...')
+---@type string[]
+local init_commands = {}
+
+local function get_lldb_commands()
+  get_rustc_sysroot(function(rustc_sysroot)
+    local script_import = 'command script import "' .. rustc_sysroot .. '/lib/rustlib/etc/lldb_lookup.py"'
+    local commands_file = rustc_sysroot .. '/lib/rustlib/etc/lldb_commands'
+    local file = io.open(commands_file, 'r')
+    if file then
+      for line in file:lines() do
+        table.insert(init_commands, line)
+      end
+      file:close()
+    end
+    table.insert(init_commands, 1, script_import)
+  end)
+end
+
+---@type string[]
+local environment = {}
+
+-- Most succinct description: https://github.com/bevyengine/bevy/issues/2589#issuecomment-1753413600
+---@param workspace_root string
+local function add_dynamic_library_paths(workspace_root)
+  compat.system({ 'rustc', '--print', 'target-libdir' }, nil, function(sc)
+    ---@cast sc vim.SystemCompleted
+    local result = sc.stdout
+    if sc.code ~= 0 or result == nil then
+      return
+    end
+    local path = (result:gsub('\n$', ''))
+    local target_path = workspace_root .. '/target/debug/deps'
+    local sep = ':'
+    local win_sep = ';'
+    if shell.is_windows() then
+      table.insert(environment, 'PATH=' .. path .. win_sep .. target_path .. win_sep .. '$PATH')
```

Comment:
I'll be honest, I am not 100% about this one. I know the first two values (`path` and `target_path`) should work, but on Windows I'm unsure whether appending `$PATH` as the last value will have the desired affect.

Reply:
I think in Windows, you would reference the `PATH` environment variable as `%PATH%`:

https://www3.ntu.edu.sg/home/ehchua/programming/howto/Environment_Variables.html#:~:text=2.3%20Using%20an%20Environment%20Variable,%22%20echo%20%25varname%25%20%22.

Reply:
Ooo yeah you're right. I do have a Windows laptop kicking around somewhere, I just haven't gone through the rigamarole of installing Rust and nvim. I presume it would need to be outside a WSL environment...